src/background.js

```diff
@@ -1,29 +1,33 @@
 window.scrNoti = window.scrNoti || {};
+const seenMessages = {};
+
+// TODO:
+// onQuit
+// mark read?
+// Update seenMessages when changing settings: simple/extended and folders
 
 //==========================================
 // On new email...
 //==========================================
 window.scrNoti.newEmailListener = async (folder, messages) => {
-  let hasNotJunk = false;
 
   if (!(await window.scrNoti.isFolderToCheck(folder))) {
     return;
   }
 
+  // Find new messages, which have not been seen yet
   if (messages && messages.messages && messages.messages.length > 0) {
     for (const message of messages.messages) {
       if (message && !message.junk) {
-        hasNotJunk = true;
-        break;
+        if (!seenMessages[folder.accountId + folder.path].has(message.id)) {
```

Comment:
I have an error here: `TypeError: seenMessages[(folder.accountId + folder.path)] is undefined`. I guess it occures it that path is not created in the "main" function.

Reply:
I do not understand that error message (and I never got it).

- Is `seenMessages` undefined? As far as I understand JavaScript, this variable should be a global variable defined at line 2.
- Is `folder.accountId` or `folder.path` undefined? These two properties are part of the Thunderbird API and should be always available.

Can you please explain in more detail, where you believe the problem lies? Thanks.

Reply:
The `.has()` part is being called on an _undefined_ object.

The easy way to fix this would be: 
```
if (seenMessages[folder.accountId + folder.path] && !seenMessages[folder.accountId + folder.path].has(message.id))
```

Reply:
I see. Does the error happen, if a new folder to watch is added in the configuration page? This is the only case I can think of, where `seenMessages[...]` is not defined for a watched folder.

Updating the `seenMessages` dictionary in the method `saveOptions` was on my list to implement, but I guess it escaped me.

Reply:
I think. Not sure. 
But anyway it's a good thing to make sure this error can't happen.

Reply:
I pulled the updating of `seenMessages` out of `window.scrNoti.main()` into the separate method `window.scrNoti.updateSeenMessages()`. How would I call this method from `saveOptions()` in `options.js`?

Reply:
`options.js` cannot call functions from `background.js`, I'm pretty sure. The only way both files can "communicate" is via storage.

Reply:
:frowning_face:

Any idea, how to update `seenMessages`, if the watched folders has been changed in the options?

Reply:
Probably by adding a "`optionsChanged`" flag in the storage, and get/clear it somewhere in `background.js` .

Reply:
Can I add a listener for such a flag?

Reply:
I'm not 100% sure, but I don't think so. 
Anyway checking the flag everytime some function is ran is not costly, it's just less pretty.

Let me know if you find another way to make both scripts communicate together. This is my very first Thunderbird extension.. I'm not pretending I'm an expert!

Reply:
Have I mentioned, that I don't know much about JavaScript and virtually nothing about Thunderbird extensions?

When I searched, how I could call a function of `background.js`, I found this:
https://stackoverflow.com/questions/5443202/call-a-function-in-background-from-popup
Does it mention anything which could be used in our case?

Reply:
Indeed, this seems to be possible after all! I can't test it until this weekend but give it a try if you want:
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/sendMessage